thrift/transport.go

```diff
@@ -49,6 +54,37 @@ func (t *readWriterTransport) Close() error {
 	return nil
 }
 
+func (t *readWriterTransport) ReadByte() (byte, error) {
+	v := t.readBuf[0:1]
+
+	var n int
+	var err error
+
+	for {
+		n, err = t.Read(v)
+		if n > 0 || err != nil {
+			break
+		}
+	}
+
+	if err == io.EOF && n > 0 {
+		err = nil
+	}
+	return v[0], err
+}
+
+func (t *readWriterTransport) WriteByte(b byte) error {
```

Comment:
Yep, since the transport wraps a `Writer`, and you can only have one goroutine writing to a `Writer` at a time, the transport calls are also serialized.